logstash_async/transport.py

```diff
@@ -3,15 +3,155 @@
 # This software may be modified and distributed under the terms
 # of the MIT license.  See the LICENSE file for details.
 
+from abc import ABC, abstractmethod
+from time import sleep
+from typing import Union
```

Comment:
In my opinion the break with python2 needs to be [strict](https://python3statement.org) now. The people had enough time (5 years) to prepare for the change. But if you really want to keep the python2 support, I will revert the changes.

Reply:
While I agree that Python2 should not be used anymore and the transition time was long enough, I also see the reality: even with this long transition, still not everyone had to chance to switch to Python3.
The main issue with dropping Python2 support is that it will hit the developers who already had no time to migrate, it will not hit the people who made this decision.
I know at least one company using this library affected.

Obviously, keeping Python2 support won't work in the long term and it doesn't make sense to do so.
But I want to keep it at least for some more time. But then finally drop Python2 support and make a major version bump.

Reply:
+ thanks for your efforts on reverting the Python3 only changes.